apiserver/facades/controller/secretbackendmanager/register.go

```diff
@@ -23,22 +23,20 @@ func Register(registry facade.FacadeRegistry) {
 }
 
 // NewSecretBackendsManagerAPI creates a SecretBackendsManagerAPI.
-func NewSecretBackendsManagerAPI(context facade.ModelContext) (*SecretBackendsManagerAPI, error) {
-	if !context.Auth().AuthController() {
+func NewSecretBackendsManagerAPI(ctx facade.ModelContext) (*SecretBackendsManagerAPI, error) {
+	if !ctx.Auth().AuthController() {
 		return nil, apiservererrors.ErrPerm
 	}
-	model, err := context.State().Model()
+	model, err := ctx.State().Model()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	serviceFactory := ctx.ServiceFactory()
+	backendService := serviceFactory.SecretBackend(model.ControllerUUID(), provider.Provider)
 	return &SecretBackendsManagerAPI{
-		watcherRegistry: context.WatcherRegistry(),
-		controllerUUID:  model.ControllerUUID(),
-		modelUUID:       model.UUID(),
-		modelName:       model.Name(),
-		backendRotate:   context.State(),
-		backendState:    state.NewSecretBackends(context.State()),
+		watcherRegistry: ctx.WatcherRegistry(),
+		backendService:  &serviceShim{backendService},
```

Comment:
Why do we need the service shim?
We need them for state when concrete structs are used, but we should not need them for services.